src/lbp.jl

```diff
@@ -0,0 +1,146 @@
+function lbp_original(bit_pattern::Array{Bool, 1})
+	sum([b * 2 ^ (length(bit_pattern) - i) for (i, b) in enumerate(bit_pattern)])
+end
+
+UNIFORM_LBP_TABLE = Dict{Array{Bool, 1}, Int}()
+UNIFORM_PATTERN_COUNT = 0
+NON_UNIFORM_PATTERN = Array{Bool, 1}()
+
+function init_uniform_lbp_params(points::Integer)
+	global UNIFORM_LBP_TABLE, NON_UNIFORM_PATTERN, UNIFORM_PATTERN_COUNT
+	
+	UNIFORM_PATTERN_COUNT::Int = 0
+	empty!(UNIFORM_LBP_TABLE)
+	NON_UNIFORM_PATTERN = zeros(Bool, points)
+	NON_UNIFORM_PATTERN[1:2:points] = true
+	UNIFORM_LBP_TABLE[NON_UNIFORM_PATTERN] = points * (points - 1) + 2
+end
+
+function lbp_uniform(bit_pattern::Array{Bool, 1})
+	global UNIFORM_LBP_TABLE, NON_UNIFORM_PATTERN, UNIFORM_PATTERN_COUNT
+	variations = sum([bit_pattern[i] != bit_pattern[i + 1] for i in 1:length(bit_pattern) - 1])
+	if variations <= 2
+		try 
+			return UNIFORM_LBP_TABLE[bit_pattern] 
+		catch KeyError 
+			UNIFORM_PATTERN_COUNT::Int += 1
+			UNIFORM_LBP_TABLE[bit_pattern] = UNIFORM_PATTERN_COUNT::Int
```

Comment:
If you take my advice below about re-using `bit_pattern`, this will have to become `UNIFORM_LBP_TABLE[copy(bit_pattern)] = ...`.